.circleci/config.yml

```diff
@@ -1,151 +1,28 @@
-          to: << parameters.s3-copy-to >>
-          arguments: --cache-control no-cache
-
-  build-docker:
-    docker:
-      - image: circleci/buildpack-deps:stretch
-    parameters:
-      mm-binary:
-        type: string
-      loadtest-binary:
-        type: string
-      docker-tag:
-        type: string
+  test:
+    executor:
+      name: default
     steps:
       - checkout
-      - setup_remote_docker
-      - run:
-          command: >
-            docker build
-            --build-arg MM_BINARY=<< parameters.mm-binary >>
-            --build-arg LOADTEST_BINARY=<< parameters.loadtest-binary >>
-            -t mattermost/mattermost-loadtest:<< parameters.docker-tag >>
-            .
-            --no-cache
-      - run:
-          command: |
-            echo $DOCKER_PASSWORD | docker login --username $DOCKER_USERNAME --password-stdin
-            docker push mattermost/mattermost-loadtest:<< parameters.docker-tag >>
-            docker logout
-            docker rmi -f mattermost/mattermost-loadtest:<< parameters.docker-tag >>
+      - run: make test
 
 workflows:
   version: 2
-  build_and_upload_and_build_push_image:
+  untagged-build:
     jobs:
-      - build-ltops-and-test
-
-      - upload-s3:
-          name: upload-s3-branches
-          s3-copy-to: s3://releases.mattermost.com/loadtest-pr/${CIRCLE_BRANCH}/
-          context: mattermost-ci-s3
-          requires:
-            - build-ltops-and-test
-          <<: *only_branches
-      - build-docker:
-          <<: *mm_binary
-          loadtest-binary: https://releases.mattermost.com/loadtest-pr/${CIRCLE_BRANCH}/mattermost-load-test.tar.gz
-          docker-tag: ${CIRCLE_SHA1}
-          context: matterbuild-docker
-          <<: *only_branches
-          requires:
-            - upload-s3-branches
-
-      - upload-s3:
-          name: upload-s3-master
-          s3-copy-to: s3://releases.mattermost.com/mattermost-load-test/
-          context: mattermost-ci-s3
-          <<: *only_master
-          requires:
-            - build-ltops-and-test
-      - build-docker:
-          <<: *mm_binary
-          loadtest-binary: https://releases.mattermost.com/mattermost-load-test/mattermost-load-test.tar.gz
-          docker-tag: master
-          context: matterbuild-docker
-          <<: *only_master
-          requires:
-            - upload-s3-master
-
-      - upload-s3:
-          name: upload-s3-tags
-          s3-copy-to: s3://releases.mattermost.com/mattermost-load-test/${CIRCLE_TAG}/
-          context: mattermost-ci-s3
-          <<: *only_v_tags
-          requires:
-            - build-ltops-and-test
-      - build-docker:
-          <<: *mm_binary
-          loadtest-binary: https://releases.mattermost.com/mattermost-load-test/${CIRCLE_TAG}/mattermost-load-test.tar.gz
-          docker-tag: ${CIRCLE_TAG}
-          context: matterbuild-docker
-          <<: *only_v_tags
-          requires:
-            - upload-s3-tags
+      - lint
+      - test
```

Comment:
Why not a build step also ?

Reply:
You mean running `make build`? I was planning on tackling that when we start doing something with the generated artifact, but 1/5 really

Reply:
I think it's good to check if the compilation succeeds. `make test` may not catch all scenarios. But if you plan to do tackle that soon then I'm fine.